### PR TITLE
ENH: add `create_diagonal`

### PR DESCRIPTION
**Pull request by lucascolley**
Closes gh-15

### CODE REVIEW

**Comment by lucascolley** [comment]
any comments appreciated @j-bowhay @mdhaber !

File: tests/test_funcs.py [hunk 1]
```diff
@@ -112,6 +112,30 @@ def test_combination(self):
         assert_allclose(cov(y, xp=xp), xp.asarray(2.144133), rtol=1e-6)
 
 
+class TestCreateDiagonal:
+    def test_1d(self):
```
Comment (j-bowhay) [inline tests/test_funcs.py:116]: Could we use the test from SciPy? I think it is stronger and helped to root out an issue in the original implementation?

**Review by j-bowhay — COMMENTED** [review]
Implementation looks good, a couple of questions around the tests

File: tests/test_funcs.py (hunk 1 above)
Comment (lucascolley) [reply tests/test_funcs.py:116]: Yep let's do that - could also contribute it back to NumPy!

**Review by j-bowhay — APPROVED** [review]
Looks good, thanks!

**Comment by lucascolley** [comment]
rendered docs look good to me, thanks for the review Jake!

**Pull request merged**